### ISSUE TITLE
Response on SystemRequest (TRAFFIC_MESSAGE_CHANNEL) returns always GENERIC_ERROR

### ISSUE THREAD

**Issue by fkattner** [smartdevicelink/sdl_ios#324]
When trying to workaround the missing imports  

The file SmartDeviceLink.h is missing some imports. It should contain:
#import “SDLSystemRequest.h”
#import “SDLSystemRequestResponse.h”

reported in  https://github.com/smartdevicelink/sdl_ios/issues/323

The response on SystemRequest (TRAFFIC_MESSAGE_CHANNEL) returns always "GENERIC_ERROR"

The single steps carried out are 
•         Reveiving trafficdata from Nokia HERE 
•         Send the data via PutFile to Sync
•         Send SystemRequest (TRAFFIC_MESSAGE_CHANNEL) to Sync
All responses return an success except for the SystemRequest.

Same approach works OK for Android.

**Comment by joeljfischer** [comment]
Hi @fkattner, thanks for reporting this and #323. If you can, try to reproduce the issue you're seeing against this branch: https://github.com/smartdevicelink/sdl_ios/tree/feature/issue_323_fix_systemrequest_unavailable

**Comment by fkattner** [comment]
Thanks will try that.

Regards,
Frank

Von: Joel Fischer [mailto:notifications@github.com]
Gesendet: Wednesday, 16. December 2015 17:00
An: smartdevicelink/sdl_ios
Cc: Kattner, Frank (F.)
Betreff: Re: [sdl_ios] Response on SystemRequest (TRAFFIC_MESSAGE_CHANNEL) returns always GENERIC_ERROR (#324)


Hi @fkattner<https://github.com/fkattner>, thanks for reporting this and #323<https://github.com/smartdevicelink/sdl_ios/issues/323>. If you can, try to reproduce the issue you're seeing against this branch: https://github.com/smartdevicelink/sdl_ios/tree/feature/issue_323_fix_systemrequest_unavailable

**Comment by ricobeck** [comment]
@joeljfischer: I'm the developer who's trying to achieve this. I've tried the fixes from the the mentioned branch but the changes are only what Ii already did to make it compile.

So the problem persists and I've tried several constellations. There is one case that ends with a success: If I use the Android implementation and send a static TMC file (with success) and then use the iOS SDK and send the same static file again the response for the system request is also SUCCESS.

I know that's not very helpful but maybe interesting. If you have any hints what I could try or need implementations details I'll answer quickly.
